### PR TITLE
MTD simulation update

### PR DESCRIPTION
**Pull request by casarsa**
BTL code:
 - removed the time of flight subtraction;
 - fixed the assignment of the DataFrame buckets;
 - added correlations between the two times that are read out.

ETL code:
 - removed the time of flight subtraction;
 - removed the linear interpolation of times in the same cell;
 - fixed the assignment of the DataFrame buckets;
 - added a time smearing depending on module eta.

The complete workflow has been successfully tested on a SingleMuon and TTbar sample with runTheMatrix.

### CODE REVIEW

**Comment by kpedro88** [comment]
please test workflow 22434.0

File: SimFastTiming/FastTimingCommon/src/ETLElectronicsSim.cc [hunk 1]
```diff
@@ -68,6 +90,11 @@ void ETLElectronicsSim::runTrivialShaper(ETLDataFrame &dataFrame,
       newSample.set(chargeColl[it] > adcThreshold_MIP_,false,tdc_time,adc);
       dataFrame.setSample(it,newSample);
 
+      //std::cout << it << " " << chargeColl[it] << " " << toa[it] << std::endl;
```
Comment (kpedro88) [inline SimFastTiming/FastTimingCommon/src/ETLElectronicsSim.cc]: delete commented-out code and extra lines

**Comment by casarsa** [comment]
@kpedro88 I've tested the workflow 22434.0:
22434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D24_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D24+RecoFullGlobal_2023D24+HARVESTFullGlobal_2023D24 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Wed Aug  1 16:30:05 2018-date Wed Aug  1 16:09:29 2018; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed

File: SimFastTiming/FastTimingCommon/src/ETLElectronicsSim.cc (hunk 1 above)
Comment (casarsa) [reply SimFastTiming/FastTimingCommon/src/ETLElectronicsSim.cc]: Thank you, I missed this. Done.

**Comment by fabiocos** [comment]
@lgray FYI

**Comment by kpedro88** [comment]
+upgrade

File: SimFastTiming/FastTimingCommon/src/BTLElectronicsSim.cc [hunk 2]
```diff
@@ -82,13 +86,18 @@ void BTLElectronicsSim::run(const mtd::MTDSimHitDataAccumulator& input,
...
+      finalToA1 += smearing_stat_thr1;
+      finalToA2 += smearing_stat_thr1 + smearing_stat_thr2;
 
 
-      // --- Add in quadrature the uncertainty due to the SiPM timing resolution:
-      sigma2_tot_thr1 += SPTR2_/TimeThreshold1_;
-      sigma2_tot_thr2 += SPTR2_/TimeThreshold2_;
+      // --- Uncertainty due to the SiPM timing resolution:
+      float sigma2_tot_thr1 = SPTR2_/TimeThreshold1_;
+      float sigma2_tot_thr2 = SPTR2_/TimeThreshold2_;
```
[the remark this answers is not in the document]
Comment (casarsa) [reply SimFastTiming/FastTimingCommon/src/BTLElectronicsSim.cc]: I've moved these lines down and hopefully improved the readability of the code.

**Comment by kpedro88** [comment]
please test workflow 22434.0

**Comment by casarsa** [comment]
@kpedro88 Workflow 22434.0 tested:
22434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D24_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D24+RecoFullGlobal_2023D24+HARVESTFullGlobal_2023D24 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Aug  6 11:31:26 2018-date Mon Aug  6 11:07:12 2018; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**